### PR TITLE
many_cubes: Add no automatic batching and generation of different meshes

### PR DESCRIPTION
**Pull request by superdump**
# Objective

- Enable stressing of more of the material mesh entity draw code paths

## Solution

- Support generation of a number of different mesh assets from the built-in primitives, and select randomly from them. This breaks batches based on different meshes.
- Support disabling automatic batching. This skips the batching cost at the expense of stressing render pass draw command encoding.
- Support enabling directional light cascaded shadow mapping - this is commonly a big source of slow down in normal scenes.

### CODE REVIEW

File: examples/stress_tests/many_cubes.rs [hunk 1]
```diff
@@ -262,6 +278,108 @@ fn init_materials(
...
+
+    let mut meshes = Vec::with_capacity(capacity);
+
+    let mut radius_rng = StdRng::seed_from_u64(42);
+    let mut variant = 0;
+    meshes.extend(
+        std::iter::repeat_with(|| {
+            let radius = radius_rng.gen_range(0.25f32..=0.75f32);
+            let (handle, transform) = match variant % 15 {
+                0 => (
```
Comment (rparrett) [inline examples/stress_tests/many_cubes.rs:292]: Wouldn't you want to only vary the uniqueness of the mesh and not the overall triangle count? Couldn't you use a bunch of separate instances of same-size cubes?

File: examples/stress_tests/many_cubes.rs [hunk 2]
```diff
@@ -262,6 +278,108 @@ fn init_materials(
...
 }
 
+fn init_meshes(args: &Args, assets: &mut Assets<Mesh>) -> Vec<(Handle<Mesh>, Transform)> {
+    let capacity = args.mesh_count.max(1);
+
+    let mut meshes = Vec::with_capacity(capacity);
+
+    let mut radius_rng = StdRng::seed_from_u64(42);
+    let mut variant = 0;
+    meshes.extend(
```
[the remark this answers is not in the document]
Comment (superdump) [reply examples/stress_tests/many_cubes.rs:288]: No reason. Just how I think. :) I can change it.

File: examples/stress_tests/many_cubes.rs (hunk 1 above)
Comment (superdump) [reply examples/stress_tests/many_cubes.rs:292]: I could, but I'd like to use a variety of meshes as well. As they are randomly distributed, the per-frame draws should be fairly consistent, if that's what you're concerned about?

File: examples/stress_tests/many_cubes.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (superdump) [reply examples/stress_tests/many_cubes.rs:288]: Done.

File: examples/stress_tests/many_cubes.rs (hunk 1 above)
Comment (rparrett) [reply examples/stress_tests/many_cubes.rs]: The concern is mainly "changing more than one variable in an experiment."
  
This switch adds a bunch of triangles to the scene, in addition to making the mesh assets unique. So it seems like like comparisons between runs where this switch is enabled and not enabled would not be useful.

I had assumed based on:
> This breaks batches based on different meshes.

That just making the assets unique was the main goal.

If that's not the case or not a worry, then I guess I'm not bothered.

Comment (superdump) [reply examples/stress_tests/many_cubes.rs]: My goal was not really to compare single mesh vs many meshes, rather just to create a scene with a bunch of different meshes. The benchmarking is done by using the same configuration of the stress test with two versions of the code.

File: examples/stress_tests/many_cubes.rs [hunk 3]
```diff
@@ -136,23 +151,40 @@ fn setup(
...
                 }
+                if args.no_automatic_batching {
+                    cube.insert(NoAutomaticBatching);
+                }
             }
 
             // camera
             commands.spawn(Camera3dBundle::default());
+            // Inside-out box around the meshes onto which shadows are cast (though you cannot see them...)
+            commands.spawn((
```
Comment (IceSentry) [inline examples/stress_tests/many_cubes.rs:174]: I'm not sure I understand what this does.

**Review by IceSentry — APPROVED** [review]

File: examples/stress_tests/many_cubes.rs [hunk 4]
```diff
@@ -262,6 +312,103 @@ fn init_materials(
     materials
 }
 
+fn init_meshes(args: &Args, assets: &mut Assets<Mesh>) -> Vec<(Handle<Mesh>, Transform)> {
+    let capacity = args.mesh_count.max(1);
+
+    let mut radius_rng = StdRng::seed_from_u64(42);
```
Comment (mockersf) [inline examples/stress_tests/many_cubes.rs:318]:
```suggestion
    // We're seeding the PRNG here to make this example deterministic for testing purposes.
    // This isn't strictly required in practical use unless you need your app to be deterministic.
    let mut radius_rng = ChaCha8Rng::seed_from_u64(42);
```

**Pull request merged**